### ISSUE TITLE
remove cross origin isolation requirement

### ISSUE THREAD

**Issue by fredericrous** [DallasHoff/sqlocal#37]
I read from the following blog post from Notion that it is possible to remove the need for cross origin isolation:
https://www.notion.so/blog/how-we-sped-up-notion-in-the-browser-with-wasm-sqlite#stumbling-block-#1-cross-origin-isolation

could sqlocal implement this?

**Comment by DallasHoff** [comment]
That post says that Notion bypassed the requirement for cross origin isolation by using [this origin trial](https://developer.chrome.com/origintrials/#/view_trial/303992974847508481). If you want to do the same, you must register your site, but keep in mind that this is only a temporary solution and only works for Chrome and Edge.

**Comment by fredericrous** [comment]
hey, thanks for the reply. 

I did not sent the correct hash link, sorry about that: https://www.notion.so/blog/how-we-sped-up-notion-in-the-browser-with-wasm-sqlite#stumbling-block-#3-the-alternative

Basically in the article Notion says they tried few alternatives and finally picked "OPFS SyncAccessHandle Pool VFS"

They didn't do much, sqlite already has a solution, documented at

https://sqlite.org/wasm/doc/trunk/persistence.md#vfs-opfs-sahpool

so basically the initialization of the db is as follow, if I understood the doc correctly

```js
const PoolUtil = await sqlite3.installOpfsSAHPoolVfs();
const db = new PoolUtil.OpfsSAHPoolDb('/filename');
```

**Comment by DallasHoff** [comment]
I decided not to use the SAH VFS because it has serious drawbacks compared to the other OPFS VFS:

1. It does not allow the same database to have multiple open connections. This means that if your app is open in multiple tabs, only one tab will be able to access the database and the others will error. To get around this, a SharedWorker can be used, like the Notion article describes, but this has the drawback that SharedWorkers have limited browser support on Android. The other VFS has full support in all modern browsers.
2. It does not store the database in a standard sqlite3 file. This means that the database cannot be exported from your app. `getDatabaseFile` would not work, and you would not be able to back up the database to a server or allow the user to download it.
